packages/backend/src/services/warehouseClients/PostgresWarehouseClient.ts

```diff
@@ -44,4 +123,66 @@ export default class PostgresWarehouseClient implements QueryRunner {
     async test(): Promise<void> {
         await this.runQuery('SELECT 1');
     }
+
+    async getSchema(
+        requests: {
+            database: string;
+            schema: string;
+            table: string;
+            columns: string[];
+        }[],
+    ) {
+        const { databases, schemas, tables } = requests.reduce<{
+            databases: Set<string>;
+            schemas: Set<string>;
+            tables: Set<string>;
+        }>(
+            (acc, { database, schema, table }) => ({
+                databases: acc.databases.add(`'${database}'`),
+                schemas: acc.schemas.add(`'${schema}'`),
+                tables: acc.tables.add(`'${table}'`),
+            }),
+            {
+                databases: new Set(),
+                schemas: new Set(),
+                tables: new Set(),
+            },
+        );
+        const query = `
```

Comment:
This will return more tables than expected I think. If I have four tables in my db:
1. `foo.bar`
2. `qux.bar`
3. `foo.baz`
4. `qux.baz`

And then I request only `foo.bar` and `qux.baz` then my query is:
```sql
            SELECT table_catalog,
                    table_schema,
                    table_name,
                    column_name,
                    data_type
             FROM information_schema.columns
             WHERE ...
             AND table_schema IN ('foo', 'qux')
             AND table_name IN ('bar', 'baz')
```

which will return all 4 tables. Not only the two I selected.

Reply:
Yup, I wrote the query like that for simplicity. I thought the amount of false positives would be low and they have no impact on the use of the catalog. In your example, when we are mapping the dbt model types the `qux.bar` and  `foo.baz` would be ignored.

I can change the query to be more specific if you want.

Reply:
I think the SQL is a good idea. 

But should we just iterate `requests` instead of `rows` in the lines below, so we don't return unexpected results?

Reply:
If we are going to have really specific query all the results will match the data in the `requests` so no reason to loop over the results since it's duplicated work. Also, even if there were extra results there is no downside to return then since they will be ignored.

Reply:
I don't think we should change the query, it's a great idea. I'm just less convinced about the safety of returning extra keys. The receiving function could easily decide to iterate the keys.

Reply:
change logic to filter out the extra resultsw